sdk/servicebus/service-bus/src/core/batchingReceiver.ts

```diff
@@ -92,7 +92,7 @@ export class BatchingReceiver extends MessageReceiver {
         const sessionError = context.session && context.session.error;
         let error = new MessagingError("An error occurred while receiving messages.");
         if (sessionError) {
-          error = translate(sessionError);
+          error = translate(sessionError) as MessagingError;
```

Comment:
Casting this way is useful if down the line we want to access properties on the `error` object that we know are available only on the `MessagingError` and not `Error`.

But the case here is not so. We are adding this cast as a means to get away from the error TS gives that `Error | MessagingError` is not assignable to `MessagingError`.

The right approach here would be to type the error object during declaration as `Error | MessagingError`.

Same applies to a few other cases in this PR.

cc @chradek for any additional thoughts here

Reply:
All changes here are because we want to access `code` and `retryable` properties which appear on `MessagingError`, so the reasoning seems fair.
The typing union would't work because the properties are not common to both (iirc while working on this locally).

Reply:
I agree that it's more correct for `error` should be declared as `Error | MessagingError` since that's what `translate` will actually return.

Related to that somewhat, you could also instantiate the default `MessagingError` later on if the `error` doesn't exist to avoid creating an error that will likely not be used (I'm assuming the `sessionError` will be available more often than not in the `onSessionError` handler).

Reply:
@ramya0820 Sorry, when I wrote my response your comment wasn't showing yet.

It doesn't look like either `code` or `retryable` are used in the `onSessionError` handler, so there's no need to case to `MessagingError` inside the handler.

Reply:
Tested on local - in general declaring type as `Error | MessagingError` wouldn't suffice, and we'll still need the `as MessagingError` casting to work with TypeScript and current code.

Only changes to be updated seems to be the 2 declarations in this file - all other places we access code and/or retryable properties and require casting.

About setting default error instance later - agree, updated

Reply:
Yes, I had noticed that about these 2 instances - I meant in general all of the changes in this PR were for accessing the properties, so agree with updates to these 2 instances to be more focused on getting around TypeScript only~

Reply:
Although, would be have been nicer if we didn't have to require the casting to access the properties if possible. Was there a reason why Error had to be a possible return value?
(Oops, looks like the docs may need to be updated on the utility as well)

Reply:
> Was there a reason why Error had to be a possible return value?

Yes, the PR #6673 has all the relevant discussions around the changes to the MessagingError class as well as for the `translate()` function.

> Looks like the docs may need to be updated on the utility as well - I mean on the translate utility in @azure/core-amqp

They were updated as part of #6673 already to the below:

> Translates the AQMP error received at the protocol layer or a SystemError into a MessagingError. All other errors are returned unaltered.

Reply:
Got it.
About the docs, I was looking at the line below that

` * @returns {MessagingError} MessagingError object. `

Reply:
Ah, that's a good catch. Please feel free to send a PR to update the return section of the docs in the master branch.